QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java

```diff
@@ -581,150 +573,136 @@ void setPanelFragment(int side, Fragment fragment) {
+                        leftDrawer, /* DrawerLayout object */
+                        actionbar.getWrappedToolbar(), /* nav drawer icon to replace 'Up' caret */
+                        R.string.hint_drawer_open, /* "open drawer" description */
+                        R.string.hint_drawer_close /* "close drawer" description */
+                ) {
+                    /**
+                     * Called when a drawer has settled in a completely closed state.
+                     */
+                    public void onDrawerClosed(View drawerView) {
+                        updateBufferRead();
+                        updateBufferRead();
+                        ((BufferFragment) manager.bufferFragment).finishActionMode();
+                        setTitleAndMenu();
+                    }
+
+                    /**
+                     * Called when a drawer has settled in a completely open state.
+                     */
+                    public void onDrawerOpened(View drawerView) {
+                        manager.closeDrawer(Side.RIGHT);
+                        setTitleAndMenu();
+                    }
+
+                    public void onDrawerSlide(View drawerView, float slideOffset) {
+                        if (manager.isDrawerVisible(Side.RIGHT))
+                            manager.closeDrawer(Side.RIGHT);
+                        super.onDrawerSlide(drawerView,0);
+                    }
+                });
+                leftDrawer.setDrawerListener((extensibleDrawerToggle.getDrawerListener()));
+            } else if (Build.VERSION.SDK_INT>=Build.VERSION_CODES.LOLLIPOP) {
+                getWindow().setStatusBarColor(getResources().getColor(R.color.primary_dark));
             }
         }
 
-        public int getOpenDrawers() {
-            if (hasDrawer)
-                return getOpenDrawers(extensibleDrawerToggle.drawer);
-            else
-                return Gravity.NO_GRAVITY;
-        }
+        private void onDrawerButtonClicked() {
+            if (manager.getLockMode(Side.RIGHT)!=DrawerLayout.LOCK_MODE_UNLOCKED)
+                return;
 
-        public int getOpenDrawers(DrawerLayout drawer) {
-            if (!hasDrawer || drawer.isDrawerVisible(Gravity.START)) {
-                return Gravity.START;
-            } else if (drawer.isDrawerVisible(Gravity.END)) {
-                return Gravity.END;
+            if (isDrawerVisible(Side.RIGHT)) {
+                closeDrawer(Side.RIGHT);
             } else {
-                return Gravity.NO_GRAVITY;
+                openDrawer(Side.RIGHT);
             }
         }
 
-        public void closeDrawerOpposite(View drawerView) {
-            if (hasDrawer && drawerView.getTag()=="LEFT")
-                closeDrawer(Gravity.START);
-            else if (hasDrawer)
-                closeDrawer(Gravity.END);
-        }
+        private void closeDrawer(Side side) {
+            if (side==Side.RIGHT||side==Side.BOTH) {
```

Comment:
The left drawer closer has a null check right below, does the right side not need that?